staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	yaml "github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+
+	"k8s.io/apiserver/pkg/server/options/encryptionconfig"
+)
+
+// EncryptionProviderOverrides is used for passing parsed information information from CLI flag to storageConfig
+type EncryptionProviderOverrides struct {
+	TransformerOverrides *map[schema.GroupResource]value.Transformer
+}
+
+func (e EncryptionProviderOverrides) Set(filepath string) error {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return fmt.Errorf("error opening encryption provider configuration file %q: %v", filepath, err)
+	}
+	defer f.Close()
+
+	if err := ConfigToTransformerOverrides(f, e.TransformerOverrides); err != nil {
```

Comment:
We generally don't parse when flags are set.  We parse afterwards in a concrete step in the initialization code.  This method is probably just a function invoked out of this package from somewhere in the generic api server startup code.